ProcessLib/VectorMatrixAssembler.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_VECTORMATRIXASSEMBLER_H_
+#define PROCESSLIB_VECTORMATRIXASSEMBLER_H_
+
+#include <vector>
+#include "NumLib/NumericsConfig.h"
+#include "AbstractJacobianAssembler.h"
+
+namespace NumLib
+{
+class LocalToGlobalIndexMap;
+}  // NumLib
+
+namespace ProcessLib
+{
+class LocalAssemblerInterface;
+
+//! Utility class used to assemble global matrices and vectors.
+//!
+//! The methods of this class get the global matrices and vectors as input and
+//! pass only local data on to the local assemblers.
+class VectorMatrixAssembler final
+{
+public:
+    explicit VectorMatrixAssembler(
+        std::unique_ptr<AbstractJacobianAssembler>&& jacobian_assembler);
+
+    void assemble(std::size_t const mesh_item_id,
```

Comment:
//! Assembles M, K, and b.  Jacobian is not assembled here, see assembleWithJacobian().